sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2795,7 +2795,24 @@ def _teardown(self,
         cluster_name = handle.cluster_name
         # Check if the cluster is owned by the current user. Raise
         # exceptions.ClusterOwnerIdentityMismatchError
-        backend_utils.check_owner_identity(cluster_name)
+        yellow = colorama.Fore.YELLOW
+        reset = colorama.Style.RESET_ALL
+        try:
+            backend_utils.check_owner_identity(cluster_name)
+        except exceptions.ClusterOwnerIdentityMismatchError as e:
+            if purge:
+                logger.error(e)
+                verb = 'terminate' if terminate else 'stop'
+                verbed = 'terminated' if terminate else 'stopped'
+                logger.warning(
+                    f'{yellow}Purge (-p/--purge) is '
+                    f'set, ignoring the above error and proceeding to attempt '
+                    f'to {verb} the cluster.{reset}\n{yellow}It is the '
+                    'user\'s responsibility to ensure that this '
+                    f'cluster is actually {verbed} on the cloud.{reset}')
```

Comment:
One way might be that we set the `identity_mismatch = True` and skip the cluster status refresh if the flag is set. Wdyt?